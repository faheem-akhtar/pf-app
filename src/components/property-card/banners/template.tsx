import { FunctionComponent } from 'react';

import { domClassMerge } from 'helpers/dom/class-merge';

import styles from '../property-card.module.scss';
import { PropertyCardBannersTemplatePropsInterface } from './template-props.interface';

export const PropertyCardBannersTemplate: FunctionComponent<PropertyCardBannersTemplatePropsInterface> = ({
  banners = [],
  showBanners = true,
}): JSX.Element | null => {
  if (!banners.length) {
    return null;
  }

  return (
    <ul
      className={domClassMerge(styles['banner-list'], {
        [styles['banner-list--hidden']]: !showBanners,
      })}
    >
      {banners.map(({ text, colorClass }, index) => (
        <li key={index} className={domClassMerge(styles.banner, styles[colorClass])}>
          {text}
        </li>
      ))}
    </ul>
  );
};
