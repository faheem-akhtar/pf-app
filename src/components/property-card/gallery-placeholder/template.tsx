import { FunctionComponent } from 'react';

import styles from './template.module.scss';

export const PropertyCardGalleryPlaceholderTemplate: FunctionComponent = () => (
  <div className={styles.container} data-testid='gallery-placeholder-image'>
    <svg className={styles.svg} viewBox='73 47 113 90'>
      <g>
        <path
          d='M186.39489,56.7832365 L186.39489,128.216763 C186.39489,130.801529 185.474537,133.014265 183.63389,134.854912 C181.793243,136.695559 179.580507,137.615912 176.995741,137.615912 L83.0042586,137.615912 C80.4194928,137.615912 78.2067571,136.695559 76.3661101,134.854912 C74.5254632,133.014265 73.6051103,130.801529 73.6051103,128.216763 L73.6051103,56.7832365 C73.6051103,54.1984708 74.5254632,51.985735 76.3661101,50.1450881 C78.2067571,48.3044411 80.4194928,47.3840882 83.0042586,47.3840882 L176.995741,47.3840882 C179.580507,47.3840882 181.793243,48.3044411 183.63389,50.1450881 C185.474537,51.985735 186.39489,54.1984708 186.39489,56.7832365 Z M176.995741,54.9034069 L83.0042586,54.9034069 C82.4951185,54.9034069 82.0545334,55.0894513 81.6825034,55.4614813 C81.3104733,55.8335113 81.1244289,56.2740964 81.1244289,56.7832365 L81.1244289,128.216763 C81.1244289,128.725904 81.3104733,129.166489 81.6825034,129.538519 C82.0545334,129.910549 82.4951185,130.096593 83.0042586,130.096593 L176.995741,130.096593 C177.504882,130.096593 177.945467,129.910549 178.317497,129.538519 C178.689527,129.166489 178.875571,128.725904 178.875571,128.216763 L178.875571,56.7832365 C178.875571,56.2740964 178.689527,55.8335113 178.317497,55.4614813 C177.945467,55.0894513 177.504882,54.9034069 176.995741,54.9034069 Z M171.356252,96.2596593 L171.356252,122.577275 L88.6437476,122.577275 L88.6437476,111.298297 L107.442044,92.5 L116.841192,101.899148 L146.918467,71.8218738 L171.356252,96.2596593 Z M111.201703,73.7017034 C111.201703,76.8347333 110.105117,79.4978645 107.912002,81.6909795 C105.718887,83.8840945 103.055755,84.9806814 99.9227255,84.9806814 C96.7896957,84.9806814 94.1265645,83.8840945 91.9334495,81.6909795 C89.7403344,79.4978645 88.6437476,76.8347333 88.6437476,73.7017034 C88.6437476,70.5686736 89.7403344,67.9055424 91.9334495,65.7124274 C94.1265645,63.5193124 96.7896957,62.4227255 99.9227255,62.4227255 C103.055755,62.4227255 105.718887,63.5193124 107.912002,65.7124274 C110.105117,67.9055424 111.201703,70.5686736 111.201703,73.7017034 Z'
          fill='currentColor'
        />
      </g>
    </svg>
  </div>
);
