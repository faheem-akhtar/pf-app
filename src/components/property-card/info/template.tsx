import { useTranslation } from 'next-i18next';

import { componentsPropertyCardGetBathroomsString } from '../get-bathrooms-string';
import { propertyCardGetBedroomsString } from '../get-bedrooms-string';
import { propertySerpObfuscatedGetBathroomValue } from 'components/property/serp/obfuscated/get/bathroom-value';
import { propertySerpObfuscatedGetBedroomValue } from 'components/property/serp/obfuscated/get/bedroom-value';
import { propertySerpObfuscatedGetPropertyTypeName } from 'components/property/serp/obfuscated/get/property-type-name';

import { PropertyCardInfoTemplatePropsInterface } from './template-props.interface';

import styles from './property-card-info.module.scss';

export const PropertyCardInfoTemplate: React.FunctionComponent<PropertyCardInfoTemplatePropsInterface> = (props) => {
  const { t } = useTranslation();
  const propertyType = propertySerpObfuscatedGetPropertyTypeName(props.property);
  const bedsValue = propertySerpObfuscatedGetBedroomValue(props.property);
  const bathsValue = propertySerpObfuscatedGetBathroomValue(props.property);

  return (
    <div className={styles.container}>
      {propertyType}
      {propertyCardGetBedroomsString(t, bedsValue)}
      {componentsPropertyCardGetBathroomsString(t, bathsValue)}
    </div>
  );
};
