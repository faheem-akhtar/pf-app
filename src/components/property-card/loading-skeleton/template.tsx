import { domClassMerge } from 'helpers/dom/class-merge';

import styles from './property-card-loading-skeleton.module.scss';

export const PropertyCardLoadingSkeletonTemplate: React.FunctionComponent = () => (
  <div className={styles.container}>
    <div className={domClassMerge(styles.gallery, 'loading')} />
    <div className={styles.content}>
      <div className={domClassMerge(styles.row1, 'loading')} />
      <div className={domClassMerge(styles.row2, 'loading')} />
      <div className={domClassMerge(styles.row2, 'loading')} />
    </div>
  </div>
);
