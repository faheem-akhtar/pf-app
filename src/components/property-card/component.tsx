import { LeadMediumType } from '@propertyfinder/pf-frontend-common/dist/module/stats/types';
import { useRouter } from 'next/router';
import { useContext, useEffect, useRef, useState } from 'react';

import { useApiPropertyImages } from 'api/property-images/hook';
import { CallingAgentModalComponent } from 'components/calling-agent-modal/component';
import { ContactedPropertyContext } from 'components/contacted-property/context';
import { EmailAgentModalComponent } from 'components/email-agent-modal/component';
import { FiltersContext } from 'components/filters/context';
import { PropertyReportComponent } from 'components/property/report/component';
import { propertySerpObfuscatedGetAreaValue } from 'components/property/serp/obfuscated/get/area-value';
import { propertySerpObfuscatedGetBathroomValue } from 'components/property/serp/obfuscated/get/bathroom-value';
import { propertySerpObfuscatedGetBedroomValue } from 'components/property/serp/obfuscated/get/bedroom-value';
import { propertySerpObfuscatedGetContactOptionsList } from 'components/property/serp/obfuscated/get/contact-options-list';
import { propertySerpObfuscatedGetId } from 'components/property/serp/obfuscated/get/id';
import { propertySerpObfuscatedGetImagesCount } from 'components/property/serp/obfuscated/get/images-count';
import { propertySerpObfuscatedGetImgUrl } from 'components/property/serp/obfuscated/get/img-url';
import { propertySerpObfuscatedGetLocationTreePath } from 'components/property/serp/obfuscated/get/location-tree-path';
import { propertySerpObfuscatedGetName } from 'components/property/serp/obfuscated/get/name';
import { propertySerpObfuscatedGetPriceText } from 'components/property/serp/obfuscated/get/price-text';
import { propertySerpObfuscatedGetPropertyTypeName } from 'components/property/serp/obfuscated/get/property-type-name';
import { propertySerpObfuscatedGetPublishDateValue } from 'components/property/serp/obfuscated/get/publish-date-value';
import { propertySerpObfuscatedGetReference } from 'components/property/serp/obfuscated/get/reference';
import { propertySerpObfuscatedGetUrl } from 'components/property/serp/obfuscated/get/url';
import { SavedPropertyContext } from 'components/saved-property/context';
import { ContactedPropertyTypeEnum } from 'enums/contacted-property/type.enum';
import { LanguageCodeEnum } from 'enums/language/code.enum';
import { arrayFromRange } from 'helpers/array/from-range';
import { dateToLocale } from 'helpers/date/to-locale';
import { functionNoop } from 'helpers/function/noop';
import { useTranslation } from 'helpers/translation/hook';
import { LocaleService } from 'services/locale/service';
import { StatsService } from 'services/stats/service';
import { TealiumConversionEventFactory } from 'services/tealium/conversion-event-factory';
import { PropertySearchStatsDataPromiseForCurrentQueryContext } from 'views/property-search/stats-data-promise-for-current-query/context';

import { PropertyShareComponent } from '../property/share/component';
import { propertyCardBannersGetBanners } from './banners/get-banners';
import { PropertyCardComponentPropsType } from './component-props.type';
import { PropertyCardMenuContentTemplate } from './menu/content/template';
import { PropertyCardMenuModalComponent } from './menu/modal/component';
import styles from './property-card.module.scss';
import { PropertyCardTemplate } from './template';
import { PropertyCardTemplatePropsType } from './template-props.type';
import { usePropertyCardTrackVisibilityOnScreen } from './track-visibility-on-screen.hook';

export const PropertyCardComponent = ({
  property,
  loading,
  onSaveButtonClick,
}: PropertyCardComponentPropsType): JSX.Element => {
  const { locale } = useRouter();
  const containerRef = useRef<HTMLDivElement>(null);
  const [galleryHasBeenTouched, setGalleryHasBeenTouched] = useState(false);
  const [isPropertySaved, setIsPropertySaved] = useState(false);
  const statsDataPromise = useContext(PropertySearchStatsDataPromiseForCurrentQueryContext);
  const { t } = useTranslation();
  const propertyId = propertySerpObfuscatedGetId(property);
  const savedProperties = useContext(SavedPropertyContext);
  const contactedProperty = useContext(ContactedPropertyContext);
  const filterContext = useContext(FiltersContext);
  const contactDate = contactedProperty.data.find(
    (property) => property.propertyId === parseInt(propertyId, 10)
  )?.contactDate;

  const timeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
  const [showBanners, setShowBanners] = useState(() => true);
  const [galleryIndex, setGalleryIndex] = useState<number>(0);

  const callingAgentModalOpenRef = useRef<() => void>(functionNoop);
  const emailAgentModalOpenRef = useRef<() => void>(functionNoop);
  const menuModalOpenRef = useRef<() => void>(functionNoop);
  const socialShareOpenRef = useRef<() => void>(functionNoop);
  const propertyReportOpenRef = useRef<() => void>(functionNoop);
  const imagesResponse = useApiPropertyImages(propertyId, 'medium', galleryHasBeenTouched);

  const tealiumEvents = TealiumConversionEventFactory(property, filterContext.value, {
    isTransactionalEvent: true,
  });

  const galleryProps = {
    items: imagesResponse.ok
      ? imagesResponse.data.map((sourceUrl) => ({ sourceUrl }))
      : [
          { sourceUrl: propertySerpObfuscatedGetImgUrl(property) },
          ...arrayFromRange(0, propertySerpObfuscatedGetImagesCount(property) - 1).map(() => ({
            sourceUrl: '',
          })),
        ],
    isRtl: locale === LanguageCodeEnum.ar,
    onTouch: (): void => setGalleryHasBeenTouched(true),
  };

  const sendLead = (medium: LeadMediumType): void => {
    statsDataPromise.then((result) => {
      if (result.ok) {
        StatsService().propertyLeadClick(parseInt(propertyId), { lead: { medium, cta: 'button' } });
      } else {
        // eslint-disable-next-line no-console
        console.error('Unable to send lead because stats data failed to load');
      }
    });
  };

  const { phone, email, whatsapp } = propertySerpObfuscatedGetContactOptionsList(property);

  const bedsValue = propertySerpObfuscatedGetBedroomValue(property);
  const bathsValue = propertySerpObfuscatedGetBathroomValue(property);

  const cardTemplateProps: PropertyCardTemplatePropsType = {
    loading,
    gallery: galleryProps,
    saved: isPropertySaved,
    contactDate: contactDate && dateToLocale(contactDate, LocaleService.getLocale()),
    type: propertySerpObfuscatedGetPropertyTypeName(property),
    price: propertySerpObfuscatedGetPriceText(property),
    customTitle: propertySerpObfuscatedGetName(property),
    location: propertySerpObfuscatedGetLocationTreePath(property),
    bathrooms: typeof bathsValue === 'number' ? t('n-bathroom', { count: bathsValue }, bathsValue > 7) : bathsValue,
    bedrooms: typeof bedsValue === 'number' ? t('n-bedroom', { count: bedsValue }, bedsValue > 7) : bedsValue,
    area: propertySerpObfuscatedGetAreaValue(property),
    publishDate: `${t('Listed')} ${propertySerpObfuscatedGetPublishDateValue(property)}`,
    href: `${propertySerpObfuscatedGetUrl(property)}?ref=listing`,
    banners: propertyCardBannersGetBanners(property, t),
    showBanners,
    // TODO-FE: CX-768 enable for other markets
    utilitiesIncluded: false,

    ...(phone
      ? {
          onCallClick: (): void => {
            sendLead('phone');
            tealiumEvents.sendCallEvent();
            callingAgentModalOpenRef.current();
            contactedProperty.add(parseInt(propertyId, 10), ContactedPropertyTypeEnum.call);
          },
          phoneNumber: phone.link,
        }
      : {}),
    ...(email
      ? {
          onEmailClick: (): void => {
            sendLead('email');
            emailAgentModalOpenRef.current();
            contactedProperty.add(parseInt(propertyId, 10), ContactedPropertyTypeEnum.email);
          },
        }
      : {}),
    ...(whatsapp
      ? {
          onWhatsappClick: (): void => {
            sendLead('whatsapp');
            tealiumEvents.sendWhatsappEvent();
            contactedProperty.add(parseInt(propertyId, 10), ContactedPropertyTypeEnum.whatsApp);
          },
          whatsAppLink: whatsapp.link,
        }
      : {}),

    onSaveButtonClick: (): void => {
      onSaveButtonClick(propertyId, !isPropertySaved);
      savedProperties.toggle(propertyId);
      if (!isPropertySaved) {
        tealiumEvents.sendSavePropertyEvent();
      }
      StatsService()[isPropertySaved ? 'propertyUnsave' : 'propertySave'](parseInt(propertyId), {});
    },
    onMenuButtonClick: (): void => {
      menuModalOpenRef.current();
    },
    t,
    onGalleryIndexChange: (index: number, length: number) => {
      tealiumEvents.sendGalleryScrollEvent(propertyId, length);
      setGalleryIndex(index);
    },
    onGalleryClick: (): void => {
      window.location.href = cardTemplateProps.href;
    },
  };

  // Handles banner animation
  useEffect(() => {
    if (timeoutRef.current) {
      clearTimeout(timeoutRef.current);
    }
    if (galleryIndex === 0) {
      setShowBanners(galleryIndex === 0);
    } else {
      setShowBanners(false);
      timeoutRef.current = setTimeout(() => {
        setShowBanners(true);
      }, 3000);
    }
    return (): void => {
      if (timeoutRef.current) {
        clearTimeout(timeoutRef.current);
      }
    };
  }, [galleryIndex]);

  useEffect(() => {
    setIsPropertySaved(!!savedProperties.data.find((property) => property.propertyId === parseInt(propertyId, 10)));
  }, [savedProperties.data, propertyId]);

  usePropertyCardTrackVisibilityOnScreen(statsDataPromise, propertyId, containerRef);

  return (
    <div data-testid='list-item' ref={containerRef} className={styles.item}>
      <PropertyCardTemplate {...cardTemplateProps} />
      <EmailAgentModalComponent
        openRef={emailAgentModalOpenRef}
        property={property}
        onFormSubmitted={(email, isAlertEnabled): void => {
          tealiumEvents.sendEmailEvent();
          if (isAlertEnabled) {
            tealiumEvents.sendEmailSignupEvent(email);
          }
        }}
      />
      <CallingAgentModalComponent
        propertyId={propertyId}
        referenceId={propertySerpObfuscatedGetReference(property)}
        openRef={callingAgentModalOpenRef}
      />
      <PropertyCardMenuModalComponent closeButtonLabel={t('cta-cancel')} openRef={menuModalOpenRef}>
        <PropertyCardMenuContentTemplate
          t={t}
          reportOpenRef={propertyReportOpenRef}
          socialShareOpenRef={socialShareOpenRef}
        />
        <PropertyShareComponent property={property} t={t} openRef={socialShareOpenRef} />
        <PropertyReportComponent propertyId={propertyId} t={t} openRef={propertyReportOpenRef} />
      </PropertyCardMenuModalComponent>
    </div>
  );
};
