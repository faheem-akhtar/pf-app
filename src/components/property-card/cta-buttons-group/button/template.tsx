import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { PropertyCardCtaButtonsGroupButtonTemplatePropsInterface } from './template-props.interface';

import styles from '../property-cta-buttons.module.scss';

export const PropertyCardCtaButtonsGroupButtonTemplate: React.FunctionComponent<PropertyCardCtaButtonsGroupButtonTemplatePropsInterface> =
  (props) => {
    return (
      <ButtonTemplate
        className={styles.button}
        size={ButtonSizeEnum.small}
        href={props.href}
        componentType={ButtonComponentTypeEnum.tertiary}
        icon={{ component: props.iconComponent, position: ButtonIconPositionEnum.left }}
        onClick={(e): void => {
          e.preventDefault();
          props.onClick();
        }}
      >
        {props.label}
      </ButtonTemplate>
    );
  };
