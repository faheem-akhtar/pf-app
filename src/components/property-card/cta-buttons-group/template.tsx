import { domClassMerge } from 'helpers/dom/class-merge';

import { IconEmailTemplate } from 'components/icon/email-template';
import { IconSolidPhoneTemplate } from 'components/icon/solid-phone-template';
import { IconWhatsappTemplate } from 'components/icon/whatsapp-template';
import { PropertyCardCtaButtonsGroupButtonTemplate } from './button/template';
import { PropertyCardCtaButtonsGroupTemplatePropsInterface } from './template-props.interface';

import styles from './property-cta-buttons.module.scss';

export const PropertyCardCtaButtonsGroupTemplate: React.FunctionComponent<PropertyCardCtaButtonsGroupTemplatePropsInterface> =
  (props) => {
    const { contactOptions, loading } = props;
    const { phone, whatsapp, email } = contactOptions;

    if (loading) {
      return <div className={domClassMerge(styles.container, { loading })} />;
    }

    return (
      <div className={styles.container}>
        {phone && (
          <PropertyCardCtaButtonsGroupButtonTemplate
            href={phone.link}
            iconComponent={IconSolidPhoneTemplate}
            onClick={props.onCallClick}
            labelKey={'Call'}
          />
        )}
        {whatsapp && (
          <PropertyCardCtaButtonsGroupButtonTemplate
            href={whatsapp.link}
            iconComponent={IconWhatsappTemplate}
            onClick={props.onWhatsappClick}
            labelKey={'WhatsApp'}
          />
        )}
        {email && (
          <PropertyCardCtaButtonsGroupButtonTemplate
            href='#'
            iconComponent={IconEmailTemplate}
            onClick={props.onEmailClick}
            labelKey={'Email'}
          />
        )}
      </div>
    );
  };
