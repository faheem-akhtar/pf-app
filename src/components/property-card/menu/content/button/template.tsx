import { PropertyCardMenuContentButtonTemplatePropsInterface } from './template-props.interface';

import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';

import styles from '../property-card-menu-content.module.scss';

export const PropertyCardMenuContentButtonTemplate = ({
  icon,
  label,
  onClick,
}: PropertyCardMenuContentButtonTemplatePropsInterface): JSX.Element => (
  <ButtonTemplate
    type='button'
    className={styles.button}
    onClick={onClick}
    componentType={ButtonComponentTypeEnum.tertiary}
    size={ButtonSizeEnum.regular}
    icon={{ component: icon, position: ButtonIconPositionEnum.left }}
  >
    {label}
  </ButtonTemplate>
);
