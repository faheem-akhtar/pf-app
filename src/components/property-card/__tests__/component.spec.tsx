/**
 * @jest-environment jsdom
 */
import { mount, shallow } from 'enzyme';
import { mockUseSwr } from 'mocks/mock/use-swr';

import { CallingAgentModalComponent } from 'components/calling-agent-modal/component';
import { EmailAgentModalComponent } from 'components/email-agent-modal/component';
import { PropertyCardComponent } from '../component';
import { PropertyCardComponentPropsType } from '../component-props.type';
import { PropertyCardTemplate } from '../template';
import { PropertyMock } from 'mocks/property/mock';

const makeDefaultProps = (): PropertyCardComponentPropsType => ({
  property: PropertyMock,
  loading: false,
});

describe('PropertyCardComponent', () => {
  /**
   * Gallery
   */
  it('Initial gallery images, before gallery touched', () => {
    const defaultProps = makeDefaultProps();
    const wrapper = shallow(<PropertyCardComponent {...defaultProps} />);

    expect(wrapper.find(PropertyCardTemplate).props().gallery.items).toMatchSnapshot();
  });

  it('should load images when gallery is touched', () => {
    const defaultProps = makeDefaultProps();
    const wrapper = shallow(<PropertyCardComponent {...defaultProps} />);

    const {
      gallery: { onTouch },
    } = wrapper.find(PropertyCardTemplate).props();
    mockUseSwr(['i1', 'i2', 'i3']);
    onTouch();
    wrapper.update();
    expect(wrapper.find(PropertyCardTemplate).props().gallery.items).toMatchSnapshot();
  });

  /**
   * CTA buttons
   */
  it('should open you are calling dialog', () => {
    const defaultProps = makeDefaultProps();
    const wrapper = mount(<PropertyCardComponent {...defaultProps} />);

    const {
      ctaButtons: { onCallClick },
    } = wrapper.find(PropertyCardTemplate).props();
    const spy = (wrapper.find(CallingAgentModalComponent).props().openRef.current = jest.fn());
    onCallClick();

    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should open email modal on email click', () => {
    const defaultProps = makeDefaultProps();
    const wrapper = mount(<PropertyCardComponent {...defaultProps} />);

    const {
      ctaButtons: { onEmailClick },
    } = wrapper.find(PropertyCardTemplate).props();
    const spy = (wrapper.find(EmailAgentModalComponent).props().openRef.current = jest.fn());
    onEmailClick();

    expect(spy).toHaveBeenCalledTimes(1);
  });
});
