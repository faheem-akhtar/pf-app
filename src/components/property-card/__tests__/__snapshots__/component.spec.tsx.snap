// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCardComponent Initial gallery images, before gallery touched 1`] = `
Array [
  Object {
    "sourceUrl": "img url",
  },
  Object {
    "sourceUrl": "",
  },
  Object {
    "sourceUrl": "",
  },
]
`;

exports[`PropertyCardComponent should load images when gallery is touched 1`] = `
Array [
  Object {
    "sourceUrl": "i1",
  },
  Object {
    "sourceUrl": "i2",
  },
  Object {
    "sourceUrl": "i3",
  },
]
`;
